src/main/webapp/app/entities/room/create/room.create.component.ts

```diff
@@ -0,0 +1,189 @@
+import { AfterContentInit, Component, OnInit } from '@angular/core';
+import { IRoomCapacity } from '../../room-capacity/room-capacity.model';
+import { EntityArrayResponseType, RoomCapacityService } from '../../room-capacity/service/room-capacity.service';
+import { IRoom, Room, RoomPicture, RoomPrice } from '../room.model';
+import { RoomService } from '../service/room.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { AlertService } from '../../../core/util/alert.service';
+
+@Component({
+  selector: 'jhi-create',
+  templateUrl: './room.create.component.html',
+  styleUrls: ['./room.create.component.scss'],
+})
+export class RoomCreateComponent implements OnInit, AfterContentInit {
+  prices: (RoomPrice & { capacityError: string; priceError: string })[] = [
+    { id: null, capacity: undefined, price: 0, capacityError: '', priceError: '' },
+  ];
+
+  capacities: IRoomCapacity[] = [];
+
+  imageAsBase64: RoomPicture[] = [];
```

Comment:
Should be renamed to `imagesAsBase64` or something similar.
So it is clear that it encapsulates multiple images and not only one.